### PR TITLE
Environment variables configuration

### LINKED ISSUE #172: Environment variables configuration
**Is your feature request related to a problem? Please describe.**
We have some environment variables configured and an opportunity to create new ones in order to simplify the development flow by centralizing this kind of information:

- Database URL
- API URL
- API port
- Secrets

If I check out the project to contribute, I have no idea of which env variables are necessary to run the project.

**Describe the solution you'd like**
Implement and configure the [dotenv](https://github.com/motdotla/dotenv) library and create an `env.example` file to list the variables needed by the project.

**Describe alternatives you've considered**
Specify the environment variables that the application needs in the `readme` file, and then the developer should configure it in your development environment. This alternative is easier to implement but is not scalable

### PR DESCRIPTION
**Pull request by oicramps**
# Environment variables configuration

## Changes

- Configured [dotenv](https://github.com/motdotla/dotenv) in the server and client side
- Created an `http` interceptor to inject baseUrl based on the `API_URL` environment variable
- Separated `dev` scripts in order to set `NODE_ENV` value.

## Issue
#172

### CODE REVIEW

File: app/angular/service/authService.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-import { BASE_URL } from "../../utils/baseUrl";
 
 const authService = function ($http, $cookies) {
 	const service = {};
 
-	const BASE_URL = "http://localhost:3000"
-
 	service.login = function (credentials) {
-		return $http.post(`${BASE_URL}/users/login`, credentials).then(function (res) {
+		return $http.post(`/users/login`, credentials).then(function (res) {
```
Comment (ArthurMota9) [inline app/angular/service/authService.js]: You can add double quotes to follow the pattern here.

**Review by ArthurMota9 — CHANGES_REQUESTED** [review]
How can we put this information about the `.env` file in the `readme`?

File: app/angular/service/authService.js (hunk 1 above)
Comment (oicramps) [reply app/angular/service/authService.js]: Done!

**Review by ArthurMota9 — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,9 @@ app/css
 app/assets
 app/dist
 
+###############################################################################
+# ENV files
+###############################################################################
+.env
+.env.development
+.env.production
```
Comment (miltonbsn) [inline .gitignore:35]: Very nice you already thought about it :white_check_mark:

**Review by miltonbsn — APPROVED** [review]
Thank you very much for this contribution mate! Certainty a nice one :+1:

Your code is clear :star:
Issue and pull request well described :star:

Anxious to see your next ones :blush:

**Comment by idmarjr** [comment]
@oicramps You have approvals on this.
I think we're good to merge \o/

**Pull request merged**